test_suite/tests/derive.rs

```diff
@@ -73,6 +76,78 @@ fn struct_derive() {
     assert_type!(SelfTyped, self_typed_type);
 }
 
+#[test]
+fn phantom_types_in_structs() {
+    #[allow(unused)]
+    #[derive(TypeInfo)]
+    struct P<T> {
+        a: u8,
+        m: PhantomData<T>,
+    }
+
+    let ty = Type::builder()
+        .path(Path::new("P", "derive"))
+        .type_params(tuple_meta_type!(bool))
+        .composite(
+            Fields::named()
+                .field_of::<u8>("a", "u8")
+                .field_of::<PhantomData<bool>>("m", "PhantomData<T>"),
+        );
+
+    assert_type!(P<bool>, ty);
+}
+
+#[test]
+fn nested_phantom_types_are_derived_in_structs() {
```

Comment:
What is this actually testing in respect to the implementation here?

I'd argue that this is a core feature of the library, supporting nested custom types, independent of the `PhantomData` feature.

Reply:
Yeah I don't disagree. It's redundant.